File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/core/services/MealMenuService.java
package com.vusystem.preschool_management_backend.modules.core.services;

import com.vusystem.preschool_management_backend.modules.core.dto.request.MealMenuRequest;
import com.vusystem.preschool_management_backend.modules.core.dto.response.MealMenuResponse;

import java.time.LocalDate;
import java.util.List;

public interface MealMenuService {
    MealMenuResponse createMealMenu(MealMenuRequest request);
    
    MealMenuResponse updateMealMenu(Long id, MealMenuRequest request);
    
    MealMenuResponse getMealMenuById(Long id);
    
    List<MealMenuResponse> getMealMenusByDate(LocalDate date);
    
    List<MealMenuResponse> getMealMenusBetweenDates(LocalDate startDate, LocalDate endDate);
    
    void deleteMealMenu(Long id);
}

File: mobile/src/services/mealMenu.service.ts
import axiosInstance from '../config/api.client';

export interface MealMenuResponse {
  id: number;
  date: string;
  mealType: string;
  description: string;
  imageUrl: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface MealMenuRequest {
  date: string;
  mealType: string;
  description: string;
  imageUrl?: string | null;
}

export const mealMenuService = {
  getMealMenusBetweenDates: async (startDate: string, endDate: string): Promise<MealMenuResponse[]> => {
    const response = await axiosInstance.get(`/core/meal-menus/range`, {
      params: { startDate, endDate }
    });
    return response.data.data;
  },
  
  getMealMenuById: async (id: number): Promise<MealMenuResponse> => {
    const response = await axiosInstance.get(`/core/meal-menus/${id}`);
    return response.data.data;
  },

  createMealMenu: async (data: MealMenuRequest): Promise<MealMenuResponse> => {
    const response = await axiosInstance.post(`/core/meal-menus`, data);
    return response.data.data;
  },

  updateMealMenu: async (id: number, data: MealMenuRequest): Promise<MealMenuResponse> => {
    const response = await axiosInstance.put(`/core/meal-menus/${id}`, data);
    return response.data.data;
  },

  deleteMealMenu: async (id: number): Promise<void> => {
    await axiosInstance.delete(`/core/meal-menus/${id}`);
  }
};
